src/network/event_loop.rs

```diff
@@ -365,6 +370,55 @@ impl EventLoop {
 						"AutoNat Old status: {:#?}. AutoNat New status: {:#?}",
 						old, new
 					);
+					// check if went private or are private
+					// if so, create reservation request with relay
+					if new == NatStatus::Private {
+						info!("Autonat says we're still private.");
+						// choose relay by random
+						if let Some(relay) = self.relay_nodes.choose(&mut rand::thread_rng()) {
```

Comment:
Should we always select a different relay or should we rely the same ones? Keeping in mind that autonat can return and will return NAT status private very often in real use cases...

Reply:
since we don't have proper discovery mechanisms in place, a random relay is preferred to kinda relax traffic going through relays

once we have discovers, we should connect the "closest" ones